### PR TITLE
Move ice1 format parsing into Ice1Format helper class

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR moves all the parsing of ice1 format strings into the Ice1Parser helper class. It does not change the parsing implementation.

### CODE REVIEW

File: csharp/src/Ice/Reference.cs [hunk 1]
```diff
@@ -70,8 +70,157 @@ public static Reference Parse(string s, Communicator communicator, string? prope
...
+                {
+                    throw new InvalidConfigurationException(
+                        $"invalid value for property `{property}': `{locatorCacheTimeout}'");
+                }
+
+                preferNonSecure = communicator.GetPropertyAsBool($"{propertyPrefix}.PreferNonSecure");
+
+                property = $"{propertyPrefix}.Router";
+                IRouterPrx? router = communicator.GetPropertyAsProxy(property, IRouterPrx.Factory);
+                if (router != null)
```
Comment (pepone) [inline csharp/src/Ice/Reference.cs:195]: We can inline router declaration here
```suggestion
                if (communicator.GetPropertyAsProxy(property, IRouterPrx.Factory) is IRouterPrx router)
```

File: csharp/src/Ice/UriParser.cs [hunk 2]
```diff
@@ -50,6 +49,25 @@ internal struct ProxyOptions
         internal static IReadOnlyList<Endpoint> ParseEndpoints(string uriString, Communicator communicator) =>
             Parse(uriString, oaEndpoints: true, communicator).Endpoints;
 
+        /// <summary>Converts the string representation of an identity to its equivalent Identity struct.</summary>
+        /// <param name="path">A string [escapedCategory/]escapedName.</param>
+        /// <exception cref="FormatException">s is not in the correct format.</exception>
```
Comment (pepone) [inline csharp/src/Ice/UriParser.cs:54]:
```suggestion
        /// <exception cref="FormatException">Thrown when <c>path</c> is not in the correct format.</exception>
```

File: csharp/src/Ice/Reference.cs (hunk 1 above)
Comment (bernardnormier) [reply csharp/src/Ice/Reference.cs]: Fixed as suggested.

File: csharp/src/Ice/UriParser.cs (hunk 2 above)
Comment (bernardnormier) [reply csharp/src/Ice/UriParser.cs]: Fixed.

**Pull request merged**